ding/reward_model/trex_reward_model.py

```diff
@@ -0,0 +1,618 @@
+            while (ti == tj):
+                #pick two random demonstrations
+                ti = np.random.randint(num_demos)
+                tj = np.random.randint(num_demos)
+            #create random partial trajs by finding random start frame and random skip frame
+            si = np.random.randint(6)
+            sj = np.random.randint(6)
+            step = np.random.randint(3, 7)
+
+            traj_i = demonstrations[ti][si::step]  # slice(start,stop,step)
+            traj_j = demonstrations[tj][sj::step]
+
+            if ti > tj:
+                label = 0
+            else:
+                label = 1
+
+            self.training_obs.append((traj_i, traj_j))
+            self.training_labels.append(label)
+            max_traj_length = max(max_traj_length, len(traj_i), len(traj_j))
+
+        #fixed size snippets with progress prior
+        for n in range(num_snippets):
+            ti = 0
+            tj = 0
+            #only add trajectories that are different returns
+            while (ti == tj):
+                #pick two random demonstrations
+                ti = np.random.randint(num_demos)
+                tj = np.random.randint(num_demos)
+            #create random snippets
+            #find min length of both demos to ensure we can pick a demo no earlier
+            #than that chosen in worse preferred demo
+            min_length = min(len(demonstrations[ti]), len(demonstrations[tj]))
+            rand_length = np.random.randint(min_snippet_length, max_snippet_length)
+            if ti < tj:  # pick tj snippet to be later than ti
+                ti_start = np.random.randint(min_length - rand_length + 1)
+                # print(ti_start, len(demonstrations[tj]))
+                tj_start = np.random.randint(ti_start, len(demonstrations[tj]) - rand_length + 1)
+            else:  # ti is better so pick later snippet in ti
+                tj_start = np.random.randint(min_length - rand_length + 1)
+                # print(tj_start, len(demonstrations[ti]))
+                ti_start = np.random.randint(tj_start, len(demonstrations[ti]) - rand_length + 1)
+            traj_i = demonstrations[ti][ti_start:ti_start + rand_length:2]  # skip everyother framestack to reduce size
+            traj_j = demonstrations[tj][tj_start:tj_start + rand_length:2]
+
+            max_traj_length = max(max_traj_length, len(traj_i), len(traj_j))
+            if ti > tj:
+                label = 0
+            else:
+                label = 1
+            self.training_obs.append((traj_i, traj_j))
+            self.training_labels.append(label)
+
+        print("maximum traj length", max_traj_length)
+        return self.training_obs, self.training_labels
+
+    def train(self):
+        #check if gpu available
+        device = self.device  # torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        # Assume that we are on a CUDA machine, then this should print a CUDA device:
+        print(device)
```

Comment:
transfer print to logger.info or logger.debug

Reply:
Thx, it is done